libs/crypto/src/crypto.cpp

```diff
@@ -20,121 +20,22 @@
-  edPkey.reset(keygenRet);
-
-  array<Byte, Ed25519PrivateKeyByteSize> privKey;
-  array<Byte, Ed25519PublicKeyByteSize> pubKey;
-  size_t keyLen{Ed25519PrivateKeyByteSize};
-
-  ConcordAssertEQ(OPENSSL_SUCCESS, EVP_PKEY_get_raw_private_key(edPkey.get(), privKey.data(), &keyLen));
-  ConcordAssertEQ(keyLen, Ed25519PrivateKeyByteSize);
-  keyLen = Ed25519PublicKeyByteSize;
-  ConcordAssertEQ(OPENSSL_SUCCESS, EVP_PKEY_get_raw_public_key(edPkey.get(), pubKey.data(), &keyLen));
-  ConcordAssertEQ(keyLen, Ed25519PublicKeyByteSize);
 
-  pair<string, string> keyPair(
-      boost::algorithm::hex(string(reinterpret_cast<char*>(privKey.data()), Ed25519PrivateKeyByteSize)),
-      boost::algorithm::hex(string(reinterpret_cast<char*>(pubKey.data()), Ed25519PublicKeyByteSize)));
-
-  if (KeyFormat::PemFormat == fmt) {
-    keyPair = EdDSAHexToPem(keyPair);
-  }
-  return keyPair;
+std::pair<std::string, std::string> generateEdDSAKeyPair(const KeyFormat fmt) {
+#ifdef USE_OPENSSL
+  return openssl::generateEdDSAKeyPair(fmt);
+#endif
 }
 
-pair<string, string> EdDSAHexToPem(const std::pair<std::string, std::string>& hex_key_pair) {
-  string privPemString;
-  string pubPemString;
-
-  if (!hex_key_pair.first.empty()) {  // Proceed with private key pem file generation.
-    const auto privKey = boost::algorithm::unhex(hex_key_pair.first);
-
-    UniquePKEY ed_privKey(EVP_PKEY_new_raw_private_key(
-        NID_ED25519, nullptr, reinterpret_cast<const unsigned char*>(privKey.data()), privKey.size()));
-    ConcordAssertNE(nullptr, ed_privKey);
-
-    UniqueBIO bio(BIO_new(BIO_s_mem()));
-    ConcordAssertNE(nullptr, bio);
-
-    ConcordAssertEQ(OPENSSL_SUCCESS,
-                    PEM_write_bio_PrivateKey(bio.get(), ed_privKey.get(), nullptr, nullptr, 0, nullptr, nullptr));
-
-    const auto lenToRead = BIO_pending(bio.get());
-    std::vector<uint8_t> output(lenToRead);
-    ConcordAssertGT(BIO_read(bio.get(), output.data(), lenToRead), 0);
-    privPemString = string(output.begin(), output.end());
-  }
-
-  if (!hex_key_pair.second.empty()) {  // Proceed with public key pem file generation.
-    const auto pubKey = boost::algorithm::unhex(hex_key_pair.second);
-
-    UniquePKEY ed_pubKey(EVP_PKEY_new_raw_public_key(
-        NID_ED25519, nullptr, reinterpret_cast<const unsigned char*>(pubKey.data()), pubKey.size()));
-    ConcordAssertNE(nullptr, ed_pubKey);
-
-    UniqueBIO bio(BIO_new(BIO_s_mem()));
-    ConcordAssertNE(nullptr, bio);
-
-    ConcordAssertEQ(OPENSSL_SUCCESS, PEM_write_bio_PUBKEY(bio.get(), ed_pubKey.get()));
-
-    const auto lenToRead = BIO_pending(bio.get());
-    std::vector<uint8_t> output(lenToRead);
-    ConcordAssertGT(BIO_read(bio.get(), output.data(), lenToRead), 0);
-    pubPemString = string(output.begin(), output.end());
-  }
-  return make_pair(privPemString, pubPemString);
+std::pair<std::string, std::string> EdDSAHexToPem(const std::pair<std::string, std::string>& hex_key_pair) {
+#ifdef USE_OPENSSL
```

Comment:
add something like
#else 
#error "Openssl is the only implementation available"

Reply:
done